API.md

```diff
@@ -0,0 +1,67 @@
+# ebookmaker-web API
+
+This web interface to ebookmaker allows a more automation-friendly access
+through a simple HTML/JSON API. By including an `out` form field with value
+`json` in the POST, a JSON object is returned with structured access to the
+ebookmaker results.
+
+## Fields
+
+The API fields match the form on the web interface, which are:
+* `out` - must be `json` to get a JSON response
+* `upfile1` - uploaded file
```

Comment:
I intentionally didn't change any of the web form names for the API. This isn't ideal IMHO because in the web form no one cares or knows about them, but users will in the API. I see 3 options:
* Keep it like it is now - the API field names match the web form names.
* Rename the web form names to things like `file`, `ebook`, `title`, `author`.
* Handle both forms - one for the web (`out=html`) and one if `out=json`.

Reply:
As long as the API is documented, I don't think it matters too much if the API names match the web form names, rather than being what you would choose if writing from scratch.

Reply:
I've done option 3 -- the form and the API use new "better" field names but the script will accept the old ones in case other folks have automation against them. We should develop against the new ones however.